lib/schema_validations/active_record/validations.rb

```diff
@@ -159,13 +159,25 @@ def load_association_validations #:nodoc:
       def add_uniqueness_validation(column) #:nodoc:
         scope = column.unique_scope.map(&:to_sym)
         name = column.name.to_sym
-        validate_logged :validates_uniqueness_of, name, :scope => scope, :allow_nil => true, :if => (proc do |record|
-          if scope.all? { |scope_sym| record.public_send(:"#{scope_sym}?") }
-            record.public_send(:"#{column.name}_changed?")
-          else
-            false
-          end
-        end)
+        options = {
+          :scope => scope, :allow_nil => true, :if => (proc do |record|
+            if scope.all? { |scope_sym| record.public_send(:"#{scope_sym}?") }
+              record.public_send(:"#{column.name}_changed?")
+            else
+              false
+            end
+          end)
+        }
+        options.merge!(:case_sensitive => false) if has_case_insensitive_index?(column, scope)
```

Comment:
I also updated the code here, think this way is more safer.  Now it will add the option only when the index is case insensitive (Because the default behaviour is case sensitive)